packages/webos/signlang/SignLangDecorator.js

```diff
@@ -0,0 +1,120 @@
+import {Component} from 'react';
+import PropTypes from 'prop-types';
+import hoc from '@enact/core/hoc';
+import {forward} from '@enact/core/handle';
+import signLang from './signLang';
+
+/**
+ * SignLangDecorator is a higher-order component that adds feature for sign language
+ * to its wrapped component.
+ *
+ *
+ * Usage:
+ * ```
+ * import {Component} from 'react';
+ * import Button from '@enact/sandstone/Button';
+ * import { SignLangDecorator } from '@enact/webos/signlang';
+ *
+ * const SignLangButton = SignLangDecorator(Button);
+ *
+ * class Sample extends Component {
+ *   render () {
+ *     return(
+ *       <div>
+ *         <SignLangButton signLangId="settings_audioguide_1">Hello</SignLangButton>
+ *       </div>
+ *     );
+ *   }
+ * }
+ * ```
+ *
+ * @class SignLangDecorator
+ * @memberof webos/signlang
+ * @hoc
+ * @public
+ */
+
+const defaultConfig = {
+	signLangDelay: 0
+};
+
+const SignLangDecorator = hoc(defaultConfig, (config, Wrapped) => {
+	const {signLangDelay} = config;
+	const forwardBlur = forward('onBlur');
+	const forwardFocus = forward('onFocus');
+
+	return class extends Component {
+		static displayName = 'SignLangDecorator';
+
+		static propTypes = /** @lends webos/signlang.SignLangDecorator.prototype */ {
+			/**
+             * Unique string used in sign language.
+             *
+             * @type {String}
+             * @public
+             */
+			signLangId: PropTypes.string,
+
+			/**
+             * Additional option for sign language
+             *
+             * @type {Object}
+             * @public
+             */
+			signLangOption: PropTypes.object
+		};
+
+		constructor (props) {
+			super(props);
+
+			this.signLangDelayId = null;
+		}
+
+		requestSignLang = (focusOut) => {
+			const {signLangId, signLangOption = {}} = this.props;
+
+			if (typeof signLangId == 'string' && signLangId.length > 0) signLang(signLangId, focusOut, signLangOption);
+		};
+
+		onFocus = (ev) => {
+			forwardFocus(ev, this.props);
+
+			if (signLangDelay === 0) {
+				this.requestSignLang(false);
+			} else {
+				clearTimeout(this.signLangDelayId);
+				this.signLangDelayId = setTimeout(() => {
+					this.requestSignLang(false);
+				}, signLangDelay);
+			}
```

Comment:
How about using `Job` from `@enact/core/util`? If we memoize the function, we can do like below.
```
const signLangJob = useMemo((focusOut = false) => new Job(requestSignLang(focusOut), signLangDelay), [requestSignLang, requestSignLangDelay]);
...

const onFocus = (ev) => {
    ...
    signLangJob.start();
}
```
Because Job will stop before it starts, you don't need to call `clearTimeout` explicitly.

Reply:
Could you try,
```
const signLangJob = useRef();
signLangJob.current = new Job(requestSignLang(focusOut), signLangDelay);
```